10.2/docker-entrypoint.sh

```diff
@@ -308,10 +321,7 @@ _mysql_passfile() {
 	# the client command will use process substitution to create a file on the fly
 	# ie: --defaults-extra-file=<( _mysql_passfile )
 	if [ '--dont-use-mysql-root-password' != "$1" ] && [ -n "$MARIADB_ROOT_PASSWORD" ]; then
-		cat <<-EOF
-			[client]
-			password="${MARIADB_ROOT_PASSWORD}"
-		EOF
+		echo -e "[client]\npassword=$(printf \"%q\" "${MARIADB_ROOT_PASSWORD}")\n"
```

Comment:
This is probably worth just bringing completely into the `printf`, ala:

```suggestion
		printf '[client]\npassword=%q\n' "$MARIADB_ROOT_PASSWORD"
```

```console
$ printf '[client]\npassword=%q\n' "foo bar baz"
[client]
password=foo\ bar\ baz
```

Reply:
I'll need to check how MariaDB processes escapes on non-quoted config items.

I would of hoped for a `[ -S "${SOCKET}" ]` however it seems there's a little too much between `listen` and the `poll`/`select` loop.